pallets/manta-pay/src/test/payment.rs

```diff
@@ -695,3 +693,140 @@ fn transfer_ledger_errors_should_shut_down() {
+            vec![ALICE],
+            vec![],
+            PalletTransferPost::try_from(to_public_input_0.clone()).unwrap(),
+            AssetType::SBT
+        ));
+        assert_ok!(MantaPay::post_transaction(
+            None,
+            vec![ALICE],
+            vec![],
+            PalletTransferPost::try_from(to_public_input_1).unwrap(),
+            AssetType::SBT
+        ));
+        assert_noop!(
+            MantaPay::post_transaction(
+                None,
+                vec![],
+                vec![ALICE],
+                PalletTransferPost::try_from(to_public.clone()).unwrap(),
+                AssetType::SBT
+            ),
+            Error::<Test>::InvalidShape
+        );
+
+        // fails for fungible ledger because there is no asset registered at this asset_id
+        assert_noop!(
+            MantaPay::post_transaction(
+                None,
+                vec![ALICE],
+                vec![],
+                PalletTransferPost::try_from(to_public_input_0).unwrap(),
+                AssetType::FT
+            ),
+            Error::<Test>::InvalidSourceAccount
+        );
+        assert_noop!(
+            MantaPay::post_transaction(
+                None,
+                vec![],
+                vec![ALICE],
+                PalletTransferPost::try_from(to_public).unwrap(),
+                AssetType::FT
+            ),
+            Error::<Test>::InvalidSinkAccount
+        );
+    });
+}
+
+#[test]
+fn private_transfer_sbt() {
+    let mut rng = OsRng;
+    new_test_ext().execute_with(|| {
+        let asset_id = 10u128;
+        let mut utxo_accumulator = UtxoAccumulator::new(UTXO_ACCUMULATOR_MODEL.clone());
+
+        let ([to_private_0, to_private_1], private_transfer) =
+            test::payment::private_transfer::prove_full(
+                &PROVING_CONTEXT,
+                &PARAMETERS,
+                &mut utxo_accumulator,
+                Fp::from(asset_id),
+                // Divide by 2 in order to not exceed total_supply
+                [100, 100],
+                &mut rng,
+            );
+
+        assert_ok!(MantaPay::post_transaction(
+            None,
+            vec![ALICE],
+            vec![],
+            PalletTransferPost::try_from(to_private_0.clone()).unwrap(),
+            AssetType::SBT
+        ));
+        assert_ok!(MantaPay::post_transaction(
+            None,
+            vec![ALICE],
+            vec![],
+            PalletTransferPost::try_from(to_private_1).unwrap(),
+            AssetType::SBT
+        ));
+        assert_noop!(
+            MantaPay::post_transaction(
+                None,
+                vec![],
+                vec![],
+                PalletTransferPost::try_from(private_transfer.clone()).unwrap(),
+                AssetType::SBT
+            ),
+            Error::<Test>::AssetSpent
+        );
+
+        // Private transfer succeeds, but this does not matter as ownership of sbt is only determined by reconstructing `ToPrivate` UTXO
```

Comment:
Cannot filter out `PrivateTransfer` as it is, well, private. 

The only way to avoid this to my understanding would to be have the SBT UTXOs in different storage from `MantaPay`. 

I opted to not do this because then we would need separate `pull_ledger_diff` rpc, duplicate indexer infrastructure  and a lot of additional upstream duplicate code. 

I think allowing dummy `private_transfers` of sbts is better than the alternative

Reply:
After talking with Fransisco I think we need separate storage for items for SBT Private ledger. Unfortunately this will require a `sbt_pull_ledger_diff` rpc and indexer infrastructure.

Currently, it would only be possible to prove ownership by extracting the `to_private` extrinsic from historical state which requires block number and needs archive node.